rasn-compiler/src/generator/rasn/builder.rs

```diff
@@ -130,12 +128,12 @@ impl Rasn {
     ) -> Result<TokenStream, GeneratorError> {
         if let ASN1Value::LinkedIntValue { integer_type, .. } = tld.value {
             let formatted_value = self.value_to_tokens(&tld.value, None)?;
-            let ty = self.to_rust_title_case(&tld.associated_type);
-            if tld.associated_type == INTEGER {
+            let ty = self.to_rust_title_case(&tld.associated_type.as_str());
+            if tld.associated_type.is_builtin_type() {
```

Comment:
In all other cases, the `is_builtin_type` check does just fine. ASN.1 `INTEGER`s however, is mapped to rust's various integer types. If you only check whether the associated type is built in, a value like `constrained INTEGER (0..10) ::= 5` would be represented as a rust value of type `Integer`, whereas in other places this type would be represented as a `u8`. 
Instead, if the associated type is an `ASN1Type::Integer(i)`, you can call the `i.int_type()` to get the rust representation of the inline `INTEGER`'s integer type, override the `LinkedIntValue`'s `integer_type`, and proceed from the `integer_type.is_unbounded()` check.

Reply:
Maybe I'm missing something by relying on `LinkedIntValue`'s `integer_type` for the output type :sweat_smile: ? This exact case is tested in the only (newly) provided test. 
Either way, I've applied your suggestion and the output type is now associated with the `associated_type`'s int type. :+1:

Reply:
My bad, I had it wrong in my first comment: the integer type was fine, the only thing was missing was the representation of smaller constrained `INTEGER` values as `const` instead of lazy static. I made a small fix that changes the corresponding lines. Thank you for your contribution and sorry for the confusion on my part 😆

Reply:
Thanks for the fix! :)

Reply:
No problem, thanks for clarifying and adding your expertise to the PR! :smiley: